Rename 'cl edit-image' to 'cl docker', remove --local from 'cl run'

fabeschan opened the pull request:
Fixes #765 and #776

percyliang commented on codalab/lib/bundle_cli.py:
```diff
@@ -123,7 +124,6 @@
 
 DOCKER_IMAGE_COMMANDS = (
-    'edit-image',
     'commit-image',
```
Should remove these too, right?

percyliang commented on codalab/lib/bundle_cli.py:
```diff
@@ -1297,7 +1266,7 @@ def do_edit_image_command(self, args):
+        print >>self.stdout, 'If you would like to commit the changes, please exit the container (ctrl-D)'
         print >>self.stdout, 'and commit your changes to a new image by running:'
         print >>self.stdout, ''
         print >>self.stdout, '\tcl commit-image {} [image-tag]'.format(container_id[:8])
```
Discourage this behavior.  Instead ask them to create a Dockerfile (point to the Wiki).

percyliang replied:
Actually, I think it would make sense to be able to specify a command here (move the logic from `run --local` basically).

The flow would be something like:
- Create Dockerfile
- docker build ...
- repeat until happy:
 * cl docker ... 'python main.py ...'
 * cl docker ... (inside it, run 'pip install')
- Edit Dockerfile to include all the pip install commands
- docker build ...
- docker push ...
- cl run --request-docker-image ...

If we wanted to be slicker, we could have `cl docker` take a `Dockerfile` directly, and have an option for appending commands to the `Dockerfile`.  Thoughts?

percyliang commented on codalab/lib/bundle_cli.py:
```diff
@@ -1306,7 +1275,7 @@ def do_edit_image_command(self, args):
+        print >>self.stdout, 'If you would like to commit the changes, please commit your changes to a new'
         print >>self.stdout, 'image by running:'
         print >>self.stdout, ''
         print >>self.stdout, '\tcl commit-image {} [image-tag]'.format(container_id[:8])
```
Can you refactor to not have this duplicated?

fabeschan committed:
Remove docker image commands, rename move local run to 'cl docker'

We remove 'cl edit-image', 'cl commit-image' and 'cl push-image'
as we want to encourage users to understand docker from its
official documentation. Keep local run functionality as a way to
test run bundles locally, and to test docker image development.

Fixes #765 and #776

fabeschan commented:
@percyliang addressed PR comments

fabeschan commented:
ready for another review

percyliang commented on codalab/lib/bundle_cli.py:
```diff
@@ -1293,56 +1258,13 @@ def do_edit_image_command(self, args):
-        print >>self.stdout, '\tdocker push [image-tag]'
-        print >>self.stdout, ''
+        print >>self.stdout, 'ContainerID: ', container_id
+        print >>self.stdout, 'Local Bundle ID: ', uuid
```
ID ==> UUID

percyliang commented on codalab/lib/bundle_cli.py:
```diff
@@ -1293,56 +1258,13 @@ def do_edit_image_command(self, args):
-        print >>self.stdout, ''
-        print >>self.stdout, '\tdocker push [image-tag]'
-        print >>self.stdout, ''
+        print >>self.stdout, 'ContainerID: ', container_id
```
add space before ID

percyliang reviewed (changes requested):
This command should behave more like `docker`, which means that:
- stdout/stderr should go to the console by default
- should be able to specify 'bash' to drop into a shell
This shouldn't create a `local_bundles` directory.

The use case I imagine is that people just run `cl docker` multiple times interleaved between editing code, so this should feel very much like running a command.

Also, we need to print out instructions on what to do with missing packages and mention `Dockerfile`.

fabeschan commented:
Ya that's good idea

fabeschan commented:
> This shouldn't create a local_bundles directory.

For people who want to debug runs that create other files, this might be very useful. I suggest we keep this directory.

> Also, we need to print out instructions on what to do with missing packages and mention Dockerfile.

This shouldn't belong to the cli help printout, but rather on external documentation like the wiki/FAQ/blogposts

fabeschan committed:
cl docker is in interactive mode by default

fabeschan commented:
@percyliang thoughts?

percyliang commented on codalab/lib/bundle_cli.py:
```diff
@@ -1293,55 +1258,13 @@ def do_edit_image_command(self, args):
-        print >>self.stdout, ''
-        print >>self.stdout, '\tcl commit-image {} [image-tag]'.format(container_id[:8])
-        print >>self.stdout, ''
+        print >>self.stdout, 'Container ID: ', container_id[:12]
```
extra space after ":", same for next lines

percyliang reviewed (commented):
I tried `cl docker --request-docker-image codalab/ubuntu:1.9 date` and it put me into a shell rather than running `date`...

fabeschan commented:
@percyliang `cl docker --request-docker-image codalab/ubuntu:1.9 date` did not drop me into a shell though..
